corehq/apps/app_manager/views/utils.py

```diff
@@ -147,3 +147,59 @@ def overwrite_app(app, master_build, include_ucrs=False, report_map=None):
                 raise AppEditingError()
     wrapped_app.copy_attachments(master_build)
     wrapped_app.save(increment_version=False)
+
+
+def get_practice_mode_configured_apps(domain, mobile_worker_id=None):
+    app_ids = set()
+    apps = {app.get_id: app for app in get_apps_in_domain(domain)}
+
+    def is_set(app_or_profile):
+        if mobile_worker_id:
+            if app_or_profile.practice_mobile_worker_id == mobile_worker_id:
+                return True
+        else:
+            if app_or_profile.practice_mobile_worker_id:
+                return True
+
+    for _id, app in apps.iteritems():
+        if is_set(app):
+            app_ids.add(_id)
+        for _, profile in app.build_profiles.iteritems():
+            if is_set(profile):
+                app_ids.add(_id)
+    return [apps[_id] for _id in app_ids]
```

Comment:
having the dict of ID to app doesn't seem useful here since you just iterate through all the apps and then return a list of apps.

It would be simpler if you just had a list.